Code/src/main/java/nl/utwente/group10/haskell/type/Type.java

```diff
@@ -7,7 +7,7 @@
 /**
  * Abstract class for Haskell types. Provides an interface for common methods.
  */
-public abstract class Type extends HaskellObject implements Comparable<Type> {
+public abstract class Type extends HaskellObject implements Comparable<Type>, Cloneable {
```

Comment:
That interface is no longer needed. Its just a sad, lonely survivor of my many tries at this.